deployments/optimism/usdc/migrations/1707394874_configurate_and_ens.ts

```diff
@@ -0,0 +1,290 @@
+
+export default migration("1707394874_configurate_and_ens", {
+  prepare: async (deploymentManager: DeploymentManager) => {
+    return {};
+  },
+
+  enact: async (
+    deploymentManager: DeploymentManager,
+    govDeploymentManager: DeploymentManager
+  ) => {
+    const trace = deploymentManager.tracer();
+    const ethers = deploymentManager.hre.ethers;
+    const { utils } = ethers;
+
+    const { bridgeReceiver, comet, cometAdmin, configurator, rewards } =
+      await deploymentManager.getContracts();
+
+    const {
+      opL1CrossDomainMessenger,
+      opL1StandardBridge,
+      governor,
+      comptrollerV2,
+      COMP: mainnetCOMP,
+      USDC: mainnetUSDC,
+      CCTPTokenMessenger,
+    } = await govDeploymentManager.getContracts();
+
+    // CCTP destination domain for Optimism
+    const OptimismDestinationDomain = 2;
+
+    // ENS Setup
+    // See also: https://docs.ens.domains/contract-api-reference/name-processing
+    const ENSResolver = await govDeploymentManager.existing(
+      "ENSResolver",
+      ENSResolverAddress
+    );
+    const subdomainHash = ethers.utils.namehash(ENSSubdomain);
+    const baseChainId = (
+      await deploymentManager.hre.ethers.provider.getNetwork()
+    ).chainId.toString();
+    const newMarketObject = {
+      baseSymbol: "USDC",
+      cometAddress: comet.address,
+    };
+    const officialMarketsJSON = JSON.parse(
+      await ENSResolver.text(subdomainHash, ENSTextRecordKey)
+    );
+    if (officialMarketsJSON[baseChainId]) {
+      officialMarketsJSON[baseChainId].push(newMarketObject);
+    } else {
+      officialMarketsJSON[baseChainId] = [newMarketObject];
+    }
+
+    const configuration = await getConfigurationStruct(deploymentManager);
+
+    const setConfigurationCalldata = await calldata(
+      configurator.populateTransaction.setConfiguration(
+        comet.address,
+        configuration
+      )
+    );
+    const deployAndUpgradeToCalldata = utils.defaultAbiCoder.encode(
+      ["address", "address"],
+      [configurator.address, comet.address]
+    );
+    const setRewardConfigCalldata = utils.defaultAbiCoder.encode(
+      ["address", "address"],
+      [comet.address, opCOMPAddress]
+    );
+    const l2ProposalData = utils.defaultAbiCoder.encode(
+      ["address[]", "uint256[]", "string[]", "bytes[]"],
+      [
+        [configurator.address, cometAdmin.address, rewards.address],
+        [0, 0, 0],
+        [
+          "setConfiguration(address,(address,address,address,address,address,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint104,uint104,uint104,(address,address,uint8,uint64,uint64,uint64,uint128)[]))",
+          "deployAndUpgradeTo(address,address)",
+          "setRewardConfig(address,address)",
+        ],
+        [
+          setConfigurationCalldata,
+          deployAndUpgradeToCalldata,
+          setRewardConfigCalldata,
+        ],
+      ]
+    );
+
+    const COMPAmountToBridge = exp(3_600, 18);
```

Comment:
Curious why this amount? Was this recommended by Gauntlet?

Reply:
Yes. The Gauntlet team provided 3600 COMP for the 6 months.